### PR TITLE
AWS IAM support for Aurora Postgres and MSK

### PR DESCRIPTION
**Pull request by stfung77**
AWS IAM support for Aurora Postgres and MSK

This change allows users to use AWS IAM to authenticate for Aurora Postgres and MSK

addresses:
https://github.com/stellar/java-stellar-anchor-sdk/issues/301
https://github.com/stellar/java-stellar-anchor-sdk/issues/323

### CODE REVIEW

File: docs/02 - Contributing/F - Testing with AWS Services.md [hunk 1]
```diff
@@ -0,0 +1,88 @@
+# Testing Anchor Platform with AWS Services
```
Comment (marcelosalloum) [inline docs/02 - Contributing/F - Testing with AWS Services.md:1]: 💡  Nice doc file! Can you also add it to the table of content [here](https://github.com/stellar/java-stellar-anchor-sdk/tree/main/docs#table-of-content)?

File: platform/src/main/resources/anchor-config-defaults.yaml [hunk 2]
```diff
@@ -215,7 +215,12 @@ app-config:
     # If useSingleQueue, all events are published to a single queue
     #   (specified in eventTypeToQueue.all)
     # eventTypeToQueue - a map of the event type to the queue name messages are published to
+    # useIAM - IAM authentication for AWS MSK
+    # NOTE: MSK - use port 9098 for access from within AWS and port 9198 for public access.
+    # NOTE: AWS_ACCESS_KEY_ID, AWS_SECRET_ACCESS_KEY and AWS_REGION environment variables need to be set
```
Comment (marcelosalloum) [inline platform/src/main/resources/anchor-config-defaults.yaml:220]: 💡  Can you add these enum variables to the `.env.example` file we have?

**Review by marcelosalloum — COMMENTED** [review]
Awesome to see this landing! I've left a few comments below ⬇️ 

Also, is it possible to unit test this configuration? If it is possible and not too troublesome, we could do that in another PR to make sure we merge this sooner rather than later.

**Review by marcelosalloum — APPROVED** [review]
Please feel free to merge it after addressing my previous comments.

If something is too troublesome, we can chat about it and create an issue to be tackled later.

File: platform/src/main/resources/anchor-config-defaults.yaml [hunk 3]
```diff
@@ -258,14 +263,17 @@ data-spring-jdbc-sqlite:
...
   spring.datasource.driver-class-name: org.postgresql.Driver
-  spring.datasource.url: jdbc:postgresql://database-2-instance-1.chizvyczscs2.us-east-1.rds.amazonaws.com:5432/
-  spring.datasource.username: ${POSTGRES_USERNAME}
-  spring.datasource.password: ${POSTGRES_PASSWORD}
+  spring.datasource.type: org.stellar.anchor.platform.databaseintegration.IAMAuthDataSource
+  spring.datasource.url: jdbc:postgresql://database-aurora-iam-instance-1.chizvyczscs2.us-east-1.rds.amazonaws.com:5432/anchorplatform
+  spring.datasource.username: anchorplatform1
+  spring.datasource.hikari.max-lifetime: 840000    # 14 minutes because IAM tokens are valid for 15min
   spring.mvc.converters.preferred-json-mapper: gson
   spring.liquibase.change-log: classpath:/db/changelog/db.changelog-master.yaml
```
[the remark this answers is not in the document]
Comment (stfung77) [reply platform/src/main/resources/anchor-config-defaults.yaml:278]: IAM for this config specifically is mandatory but you can still authenticate with username/password using the other example

File: platform/src/main/resources/anchor-config-defaults.yaml [hunk 4]
```diff
@@ -215,7 +215,12 @@ app-config:
     # If useSingleQueue, all events are published to a single queue
     #   (specified in eventTypeToQueue.all)
     # eventTypeToQueue - a map of the event type to the queue name messages are published to
+    # useIAM - IAM authentication for AWS MSK
+    # NOTE: MSK - use port 9098 for access from within AWS and port 9198 for public access.
```
[the remark this answers is not in the document]
Comment (stfung77) [reply platform/src/main/resources/anchor-config-defaults.yaml:219]: a must, i'll add the amazon doc link

File: docs/02 - Contributing/F - Testing with AWS Services.md [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+
+Anchor Platform can be configured to work with the following AWS Services:
+- RDS - Aurora Postgres Database (with IAM support)
+- MSK - Managed Streaming for Kafka (with IAM support)
+- SQS - Simple Queue Service
+
+
+##Aurora Postgres (with IAM)
+
+###AWS Configuration:
```
[the remark this answers is not in the document]
Comment (stfung77) [reply docs/02 - Contributing/F - Testing with AWS Services.md:11]: oh i didnt know that, it renders fine in intelliJ

**Pull request merged**